internal/api/http/controller/registration.go

```diff
@@ -1,34 +1,67 @@
 package controller
 
 import (
+	"encoding/json"
 	"net/http"
 
 	"go.opentelemetry.io/otel/trace"
 
 	"GoEdu/internal/api/http/server"
 	"GoEdu/internal/pkg/logger"
+	"GoEdu/internal/registration/api/inprocess"
 )
 
 type registrationHTTPController struct {
-	tracer trace.Tracer
+	registrationAPI inprocess.RegistrationModuleFacade
+	tracer          trace.Tracer
 }
 
-func newRegistrationHTTPController(t trace.Tracer) *registrationHTTPController {
-	return &registrationHTTPController{tracer: t}
+func newRegistrationHTTPController(
+	t trace.Tracer,
+	rAPI inprocess.RegistrationModuleFacade,
+) *registrationHTTPController {
+	return &registrationHTTPController{tracer: t, registrationAPI: rAPI}
 }
 
 func (c *registrationHTTPController) RegistrationNew(w http.ResponseWriter, r *http.Request) {
 	logger.FromContext(r.Context()).Info("registration new")
-	w.WriteHeader(http.StatusNotImplemented)
+	cmd := inprocess.RegisterNewUserCommand{}
+
+	err := json.NewDecoder(r.Body).Decode(&cmd)
```

Comment:
in `server.gen.go` there is `NewRegistration` request body struct. Please decode the json body into this struct and then map the `RegisterNewUserCommand` manually.